From 81852b0a279ab38903caab7f1961339ec79d4826 Mon Sep 17 00:00:00 2001
From: Anatoly Myachev <anatoly.myachev@intel.com>
Date: Wed, 30 Oct 2024 14:56:52 +0100
Subject: [PATCH 11/14] Don't specify `-A x64` option and reuse cmake build
 type config for Windows (#5014)

The `-A` argument is not compatible with the Ninja generator.

Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>
---
 python/setup.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/python/setup.py b/python/setup.py
index c60dc6158..da924502b 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -378,12 +378,10 @@ class CMakeBuild(build_ext):
         cfg = get_build_type()
         build_args = ["--config", cfg]
 
+        cmake_args += [f"-DCMAKE_BUILD_TYPE={cfg}"]
         if platform.system() == "Windows":
             cmake_args += [f"-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_{cfg.upper()}={extdir}"]
-            if sys.maxsize > 2**32:
-                cmake_args += ["-A", "x64"]
         else:
-            cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
             max_jobs = os.getenv("MAX_JOBS", str(2 * os.cpu_count()))
             build_args += ['-j' + max_jobs]
 
